### PR TITLE
feat: Expose repo flags for Go, Ruby, Java and .Net apm agents

### PR DESCRIPTION
**Pull request by v1v**
Fixes https://github.com/elastic/apm-integration-testing/issues/41

## Highlights
- Expose the below arguments/flags
  - `--go-agent-repo`
  - `--ruby-agent-repo`
  - `--java-agent-repo`
  - `--dotnet-agent-repo`
- Checkout either a branch or a commit
- Neither python nor nodejs support those flags at the moment.

### CODE REVIEW

**Comment by v1v** [comment]
Jenkins run the tests

File: scripts/compose.py [hunk 1]
```diff
@@ -1247,10 +1248,15 @@ def add_arguments(cls, parser):
             default=cls.DEFAULT_AGENT_VERSION,
             help='Use Go agent version (master, 0.5, v0.5.2, ...)',
         )
+        parser.add_argument(
+            '--go-agent-repo',
+            default=cls.DEFAULT_AGENT_REPO,
```
Comment (kuisathaverat) [inline scripts/compose.py]: you've forgotten the help :)

**Comment by graphaelli** [comment]
We should be adding tests for this, if only to handle regressions.

**Comment by v1v** [comment]
Sure thing, I actually changed some UTs:
- https://github.com/elastic/apm-integration-testing/pull/473/files#diff-aeb91b03c5e397e067ce809ea239df49

But let me add some other tests to ensure those flags work as expected, I'll raise another PR with the suggested test.

Thanks Gil

**Comment by graphaelli** [comment]
I'd suggest a test where the repo doesn't match the default.

**Pull request merged**